### PR TITLE
Support docker

### LINKED ISSUE #33: Can not run any commands
Hi i am using linux and the extension does not work at all for me.

I get the following errors
```
[CMD] "/usr/bin/php" artisan cache:clear  
[ERR] Command failed: cd "/home/tecbeast/Code/path/to/project/" && "/usr/bin/php" artisan cache:clear
```

Running /usr/bin/php -v

```
PHP 7.2.11 (cli) (built: Oct  9 2018 18:14:59) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
```

am i missing something here?

### PR DESCRIPTION
**Pull request by robsontenorio**
Fix #33 

Many people are using Docker, so here is the Docker Support for this nice package.

### CODE REVIEW

File: src/Common.ts [hunk 1]
```diff
@@ -86,13 +86,22 @@ export default class Common {
...
-    let cmd = process.platform == 'win32' ?
-      // Windows command
-      `cd /d "${artisanRoot}" && ${command}` :
-      // Unix command
-      `cd "${artisanRoot}" && ${command}`
+    let cmd
+
+    if (dockerEnabled) {
+      command = `php artisan ${command}`.trim()
+      cmd = `cd ${artisanRoot} && ${dockerCommand} ${command}`
```
Comment (TheColorRed) [inline src/Common.ts]: Should `${command}` be wrapped within quotes?

**Comment by TheColorRed** [comment]
Thanks for the PR! I have added a comment to a line of the code, could you check it out and leave me some feedback?

Thanks.

File: package.json [hunk 2]
```diff
@@ -42,6 +42,16 @@
...
           "description": "Sets a custom location to the php executable"
+        },
+        "artisan.docker.enabled": {
+          "type": "boolean",
+          "default": false,
+          "description": "Should commands get executed using docker?"
+        },
+        "artisan.docker.command": {
+          "type": "string",
+          "default": null,
```
Comment (TheColorRed) [inline package.json]: Maybe the default value could be `"docker-compose exec app"`? That is assuming this is the most common command.

Comment (robsontenorio) [reply package.json]: “app” string was an example. This is not the default. Keep attribute as null is the best choice. Will force user write their own full custom commands to get container running.

File: src/Common.ts (hunk 1 above)
Comment (robsontenorio) [reply src/Common.ts]: I am not sure about your comment. I will check it ASAP

Comment (robsontenorio) [reply src/Common.ts]: @TheColorRed Tried both, and both will work, because docker containners runs linux.

**Comment by robsontenorio** [comment]
@TheColorRed Done!

**Comment by robsontenorio** [comment]
@TheColorRed works like a charm. For sure more can be improved. Lets wait for feedback.

Thanks!

**Pull request merged**